server/channels/app/platform/session.go

```diff
@@ -50,18 +51,52 @@ func (ps *PlatformService) AddSessionToCache(session *model.Session) {
 }
 
 func (ps *PlatformService) ClearUserSessionCacheLocal(userID string) {
-	if keys, err := ps.sessionCache.Keys(); err == nil {
-		var session *model.Session
-		for _, key := range keys {
-			if err := ps.sessionCache.Get(key, &session); err == nil {
-				if session.UserId == userID {
-					ps.sessionCache.Remove(key)
-					if m := ps.metricsIFace; m != nil {
-						m.IncrementMemCacheInvalidationCounterSession()
-					}
+	var toDelete []string
+	// First, we iterate over the entire session cache.
+	err := ps.sessionCache.Scan(func(keys []string) error {
+		if len(keys) == 0 {
+			return nil
+		}
+
+		toPass := make([]any, 0, len(keys))
+		for i := 0; i < len(keys); i++ {
+			var session *model.Session
+			toPass = append(toPass, &session)
+		}
```

Comment:
After looking at this pattern for too long (from the previous PR as well), I wonder whether passing a generic allocator function (`func() any`) to `GetMulti` would work and simplify a bit. Then we could allocate the values slice internally and return it along with any errors.

Reply:
Hmm .. ok I will look into it separately. https://mattermost.atlassian.net/browse/MM-60171